src/mbgl/renderer/sources/render_raster_source.cpp

```diff
@@ -44,24 +36,36 @@ void RenderRasterSource::updateTiles(const TileParameters& parameters) {
 
     if (tileURLTemplates != tileset->tiles) {
         tileURLTemplates = tileset->tiles;
-        tilePyramid.invalidateTiles();
+
+        // TODO: this removes existing buckets, and will cause flickering.
+        // Should instead refresh tile data in place.
```

Comment:
No, this was the case with the prior implementation as well, so I'd like to leave it as a followup.